src/backend/distributed/commands/call.c

```diff
@@ -101,51 +104,33 @@ CallFuncExprRemotely(CallStmt *callStmt, DistObjectCacheEntry *procedure,
 
 	CitusTableCacheEntry *distTable = GetCitusTableCacheEntry(colocatedRelationId);
 	Var *partitionColumn = distTable->partitionColumn;
+	bool colocationWithReferenceTable = false;
```

Comment:
For this part: https://github.com/citusdata/citus/blob/04e73b8ad2b89c60c5d43c615049be5f70acae06/src/backend/distributed/commands/call.c#L107..L126

Can we do something like below ? (includes some simplifications and style changes)

```c
    ShardPlacement *placement = NULL;
	bool colocationWithReferenceTable = (partitionColumn == NULL);
	if (colocationWithReferenceTable)
	{
        // review : indentation & removed the comment
		ereport(DEBUG1, (errmsg("will push down CALL for reference "
                                "tables")));        
		Assert(IsReferenceTable(colocatedRelationId));

        placement = ShardPlacementWhenColocatedWithReferenceTable(distTable);
	}
    else
    {
	    placement = ShardPlacementWhenColocatedWithDistTable(procedure, funcExpr,
 										                     partitionColumn, distTable);
    }
```

Reply:
Even though not related to this PR, I find this if check a bit implicit. I think it would make a lot of sense to have a utility function that takes a `citus cache entry` and returns `true/false` based on whether it is a reference table or not, that way we don't need to know the details of how a table is considered as a reference table, so we hide that information, and it can change in the future. If we had a utility method for that, we will only change it in one place, however with the current approach we will need to update all the codebase. 

For example, citus local tables will also have `partitionColumn` as `NULL`, so this way of checking will be incorrect I guess.

Reply:
#4123.